runtime/src/asset.rs

```diff
@@ -271,19 +258,26 @@ decl_module! {
             // Check that sender is allowed to act on behalf of `did`
             ensure!(<identity::Module<T>>::is_authorized_key(did, &sender_key), "sender must be a signing key for DID");
 
+            ensure!(!<Tokens<T>>::exists(&ticker), "token already created");
+
+            let ticker_config = Self::ticker_registration_config();
+
+            ensure!(ticker.len() <= usize::try_from(ticker_config.max_ticker_length).unwrap_or_default(), "ticker length over the limit");
```

Comment:
Instead of duplicating the logic of ticker registration, couldn't this call into register_ticker()?

Reply:
Most of the logic is unique because of the order in which it is called or due to different requirements. For example, register ticker sets ticker expiry to whatever we define in ticker_config but create token sets the expiry to none i.e the ticker never expires after token creation.

Reply:
I refactored it to remove most of the duplicate code. The rest of it is pretty conditional.